sourcery_analytics/extractors.py

```diff
@@ -107,6 +110,7 @@ class Extractor(typing.Generic[T]):
     """
 
     visitor: Visitor[typing.Optional[T]] = IdentityVisitor()
+    manager: astroid.manager.AstroidManager = astroid.manager.AstroidManager()
```

Comment:
**suggestion (performance):** Consider using a singleton for AstroidManager.

Instantiating AstroidManager directly in the Extractor class might lead to unnecessary memory usage and initialization overhead if multiple Extractor instances are created. AstroidManager is typically used as a singleton, so consider accessing it via a class method or a module-level instance.

```suggestion
    manager: astroid.manager.AstroidManager = astroid.manager.AstroidManager()
+   manager = astroid.manager.AstroidManager()
```